core/policy_test.go

```diff
@@ -150,6 +152,45 @@ func TestMessageQueuePolicy(t *testing.T) {
 		require.Error(t, err)
 		assert.Contains(t, err.Error(), "nonce 1, expected 2")
 	})
+
+	t.Run("removes sequential messages in peer blocks", func(t *testing.T) {
+		blocks := th.NewFakeBlockProvider()
+		q := core.NewMessageQueue()
+		policy := core.NewMessageQueuePolicy(blocks, 10)
+
+		msgs := []*types.SignedMessage{
+			requireEnqueue(q, mm.NewSignedMessage(alice, 1), 100),
+			requireEnqueue(q, mm.NewSignedMessage(alice, 2), 101),
+		}
+
+		root := blocks.NewBlock(0)
+		root.Height = 100
+
+		// Construct two blocks at the same height, each with one message. The messages must
+		// have been mined in nonce order. The canonical tipset block ordering
+		// is given by block ticket, which hence must match this nonce order.
+		// These blocks are constructed so that their CIDs would order them
+		// in the *opposite* order (blocks used to be ordered by CID).
+		b1 := blocks.NewBlockWithMessages(1, []*types.SignedMessage{msgs[0]}, root)
+		b2 := blocks.NewBlockWithMessages(1, []*types.SignedMessage{msgs[1]}, root)
+		b1.Ticket = []byte{0}
+		b2.Ticket = []byte{1}
+		b2.Timestamp = 0 // Tweak to force CID ordering to be opposite to ticket ordering.
+		assert.True(t, bytes.Compare(b1.Cid().Bytes(), b2.Cid().Bytes()) > 0)
+
+		err := policy.HandleNewHead(ctx, q, requireTipset(t, root), requireTipset(t, b1, b2))
+		require.NoError(t, err)
+		assert.Empty(t, q.List(alice))
+
+		// If the blocks were ordered the other way, this would fail.
```

Comment:
This comment is confusing.  I think that this test would fail at the require on line 181 if the blocks were ordered in the other way.  `HandleNewHead` would have called `RemoveNext` on `b2` first and `RemoveNext` would return [this error](https://github.com/filecoin-project/go-filecoin/blob/master/core/message_queue.go#L91).  Putting this comment here makes it look like the test would fail on line 186 which it would never get to.

Reply:
Oh I understand what you mean now.  One way to make this clearer might be to add a comment above line 181 like "If the blocks are ordered `b1, b2` then this passes" and then have this comment.

You could also change this comment to say "*When* the blocks are ordered in reverse, this *should* fail" to make it clear that the "failure" you are talking about is captured in the code block below.